### DISCUSSION TITLE
heltec v2 support

### DISCUSSION THREAD

**Discussion by cracky22** [meshtastic/firmware#2864] [Q&A] [answered]
Why is heltec v2 not supported and are there options to run the current version on it? I have 5 Heltec v2 boards and don't want to throw them away

**Comment by lachesis** [comment]
I'd like to know this too. The only things I can find are the comment on the docs page:

>Not recommended because of design issues! Support is being phased out. Use V3 in new projects.
https://meshtastic.org/docs/hardware/devices/heltec/

And a bunch of other forum posts asking why. Does anyone know what specific "design issues" these boards suffer from? I have a handful of them and I don't really want to upgrade right now.

As for how to continue to flash them, they don't appear to be in the web flasher, but their files still seem to be in the firmware bundle. If you use esptool.py, you should be able to flash with the latest meshtastic firmware. I'm testing right now, will report back.

**Reply by lachesis** [reply]
I was able to successfully flash two of these boards with latest Meshtastic. It seems to function relatively well, at least in my unconstrained environment (USB power, no battery). So far, I've tested sending & receiving messages, uplinking to MQTT, and configuring over HTTP API. So far, the only real issue I've seen is that about 2/3rds of the time, commands sent using the meshtastic CLI over Wifi fail. It works perfectly when connected to USB serial and seems to reliably uplink to MQTT.

**Reply by cracky22** [reply]
hi, could you please briefly explain to me how I can access the heltec with esptool.py, e.g. 2.2.18er can update? so without a reset

**Reply by hbarnard** [reply]
Hi which firmware did you use to flash? I've used `./device-install.sh -f firmware-heltec-v2_0-2.7.26.54e0d8d.factory.bin` which gives me viable function but http ports (all ports actually) are closed according to `nmap`, can't get in using https therefore. Only have one v2 which was a gift but don't want to bin it, e-waste etc.

**Comment by garthvh** [comment, marked as answer]
It has a ton of battery issues and the chip used crashes and behaves erratically.

**Reply by cracky22** [reply]
ok thanks

**Reply by cracky22** [reply]
Why is the button click on the Heltec v2 not detected or executed sometimes?
